### PR TITLE
Optimization of the blinding circuit

### PR DESCRIPTION
**Pull request by XuyangSong**
#72 

- [x] Hash the 6 blinded elements instead of all in vp. An improvement about blind circuit size from 2^15 to 2^13

### CODE REVIEW

File: src/circuit/blinding_circuit.rs [hunk 1]
```diff
@@ -74,22 +69,29 @@ where
...
+        //         poseidon_circuit.output_hash(composer)
+        //     })
+        //     .collect::<Vec<Variable>>();
+
+        // poseidon_circuit.reset(composer);
+        // for v in hash_vec.iter() {
+        //     poseidon_circuit.input(*v).unwrap();
+        // }
+
+        // Compress blinded elements in vp
```
Comment (simonmasson) [inline src/circuit/blinding_circuit.rs:90]:
```suggestion
        // Compress x-coordinate blinded elements in vp
```

File: src/vp_description.rs [hunk 2]
```diff
@@ -50,29 +51,35 @@ impl<CP: CircuitParameters> ValidityPredicateDescription<CP> {
...
-                poseidon.reset(&mut ());
-                for x in hash_vec.iter() {
-                    poseidon.input(*x).unwrap();
-                }
+                // poseidon.reset(&mut ());
+                // for x in hash_vec.iter() {
+                //     poseidon.input(*x).unwrap();
+                // }
+
+                // Compress blinded elements in vp
```
Comment (simonmasson) [inline src/vp_description.rs:79]:
```suggestion
                // Compress x-coordinate blinded elements in vp
```

**Review by simonmasson — APPROVED** [review]
LGTM. It reduces the (partial) blinding circuit but if we keep this blinding construction (with a tweak for the σ polynomials), we will go back to larger circuits as before.

**Comment by XuyangSong** [comment]
Not needed so far

**Pull request closed without merging**